src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs

```diff
@@ -0,0 +1,267 @@
+                {
+                    fValid &= DropAllOfKindNested(ref itemType, errors, args[i], DKind.DataEntity);
+                }
+
+                fValid &= DropAttachmentsIfExists(ref itemType, errors, args[i]);
+            }
+
+            Contracts.Assert(!itemType.IsValid || itemType.IsTable);
+            collectedType = itemType.IsValid ? itemType : DType.EmptyTable;
+            return fValid;
+        }
+
+        // Typecheck an invocation of Collect.
+        public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            var fValid = base.CheckInvocation(binding, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            Contracts.Assert(returnType.IsTable);
+
+            DType dataSourceType = argTypes[0];
+
+            // Need a collection for the 1st arg
+            DType collectionType = argTypes[0];
+            if (!collectionType.IsTable)
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg, Name);
+                fValid = false;
+            }
+
+            // Get the unified collected type on the RHS. This will generate appropriate
+            // document errors for invalid arguments such as unsupported aggregate types.
+            fValid &= TryGetUnifiedCollectedType(args, argTypes, errors, out DType collectedType);
+            Contracts.Assert(collectedType.IsTable);
+
+            // The item type must be compatible with the collection schema.
+            var fError = false;
+            returnType = DType.Union(ref fError, collectionType, collectedType, useLegacyDateTimeAccepts: true);
+            if (fError)
+            {
+                fValid = false;
+                if (!SetErrorForMismatchedColumns(collectionType, collectedType, args[1], errors))
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg);
+                }
+            }
+
+            return fValid;
+        }
+
+        // This method returns true if there are special suggestions for a particular parameter of the function.
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        public override bool TryGetDataSourceNodes(CallNode callNode, TexlBinding binding, out IList<FirstNameNode> dsNodes)
```

Comment:
>TryGetDataSourceNodes [](http://example.com/codeflow?start=29&length=21)

You can remove this (and GetIdentifierOfModifiedValue) 
They're not used here. 
(just used in more complex powerapps scenarios)

Reply:
Removed.